cmd/backfillEvents.go

```diff
@@ -0,0 +1,65 @@
+package cmd
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/makerdao/vulcanizedb/libraries/shared/logs"
+	"github.com/makerdao/vulcanizedb/utils"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+// backfillEventsCmd represents the backfillEvents command
+var backfillEventsCmd = &cobra.Command{
+	Use:   "backfillEvents",
+	Short: "BackFill events from already-checked headers",
+	Long: `Fetch and persist events from configured transformers across a range
+of headers that may have already been checked for logs. Useful when adding a
+new event transformer to an instance that has already been running and marking
+headers checked as it queried for the previous (now incomplete) set of logs.`,
+	Run: func(cmd *cobra.Command, args []string) {
+		err := backFillEvents()
+		if err != nil {
+			logrus.Fatalf("error back-filling events: %s", err.Error())
+		}
+		logrus.Info("completed back-filling events")
+	},
+}
+
+func init() {
+	rootCmd.AddCommand(backfillEventsCmd)
+	backfillEventsCmd.Flags().Int64VarP(&endingBlockNumber, "ending-block-number", "e", -1, "last block from which to back-fill events")
+
+}
+
+func backFillEvents() error {
+	if endingBlockNumber == int64(-1) {
+		return errors.New("ending block number required to back-fill events")
+	}
+
+	ethEventInitializers, _, _, exportTransformersErr := exportTransformers()
+	if exportTransformersErr != nil {
+		LogWithCommand.Fatalf("SubCommand %v: exporting transformers failed: %v", SubCommand, exportTransformersErr)
+	}
+
+	blockChain := getBlockChain()
+	db := utils.LoadPostgres(databaseConfig, blockChain.Node())
```

Comment:
Are you sure databaseConfig is actually set here? I think it's only sat as part of the backfillUrns backfill.

Reply:
I think this'll be setup by the `PersistentPreRun` setup defined in `cmd/root.go`: https://github.com/makerdao/vulcanizedb/blob/staging/cmd/root.go#L88

I think some of the confusion here might derive from the fact that we also have a `cmd` directory in `vdb-mcd-transformers`, which is where `backfillUrns` lives: https://github.com/makerdao/vdb-mcd-transformers/tree/staging/cmd

Reply:
👍